sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1530,6 +1650,18 @@ class Analyzer(
   object GlobalAggregates extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
       case Project(projectList, child) if containsAggregates(projectList) =>
+
+        def traversePlanAndCheck(plan: LogicalPlan): Unit =
```

Comment:
Can we move it to `traversePlanAndCheck` above?

Reply:
I moved all the checks to a dedicated rule.